manifests/install.pp

```diff
@@ -38,6 +38,25 @@
       source    => "/opt/puppetlabs/packages/${_unzipped_package_name}",
       require   => Class['puppet_agent::install::remove_packages'],
     }
+  } elsif $::operatingsystem == 'Solaris' and $::operatingsystemmajrelease == '11' {
+    contain puppet_agent::install::remove_packages
+
+    exec { 'puppet_agent restore /etc/puppetlabs':
+      command => 'cp -r /tmp/puppet_agent/puppetlabs /etc',
+      path    => '/bin:/usr/bin:/sbin:/usr/sbin',
+      require => Class['puppet_agent::install::remove_packages'],
+    }
+
+    exec { 'puppet_agent post-install restore /etc/puppetlabs':
+      command     => 'cp -r /tmp/puppet_agent/puppetlabs /etc',
+      path        => '/bin:/usr/bin:/sbin:/usr/sbin',
+      refreshonly => true,
+    }
+
+    $_package_options = {
+      require => Exec['puppet_agent restore /etc/puppetlabs'],
+      notify  => Exec['puppet_agent post-install restore /etc/puppetlabs'],
```

Comment:
@jpartlow We are backing up `/etc/puppetlabs/` here: https://github.com/puppetlabs/puppetlabs-puppet_agent/pull/73/files#diff-0b65fa0c1283d2894d43d73f820fe08eR46

Also the restore is needed twice because when we remove the pe-* packages all of /etc/puppetlabs is moved into a lost+found directory determined by the system. The first restore is needed before the puppet-agent package is installed because pkg expects the repo we setup to be in `/etc/puppetlabs/installer/solaris.repo/` with the new agent package. The second restore is because when you install the puppet-agent package solaris overwrites any existing user configuration in `/etc/puppetlabs/` leaving the puppet.conf in a bad state where it doesn't know its master or any other configuration that a user may have specified.